console-csharp-snippets-sample/AuthenticationHelper.cs

```diff
@@ -34,9 +35,6 @@ class AuthenticationHelper
 
         public static PublicClientApplication IdentityClientApp = new PublicClientApplication(clientIdForUser);
         public static ConfidentialClientApplication IdentityAppOnlyApp = new ConfidentialClientApplication(clientIdForApp, Constants.AuthorityUri, Constants.RedirectUriForAppAuthn, new ClientCredential(Constants.ClientSecret), new TokenCache(), new TokenCache());
-        public static string TokenForUser = null;
-        public static string TokenForApp = null;
-        public static DateTimeOffset Expiration;
```

Comment:
You should **never** keep tokens in variables. Rather use AcquireTokenSilentAsync for user tokens, and AcquireTokensForClient already leverages the app token cache.

The reason is the token can become expired, or be revoked, and your code will break.

You should **never** keep the expiration date either. MSAL's token cache is there to take care of refreshing the tokens when calling AcquireTokenSilentAsync

Reply:
Got it, thanks for illustrating this pattern.